### DISCUSSION TITLE
Simple stats daemon for RE2 servers!

### DISCUSSION THREAD

**Discussion by enp2s0** [redeclipse/base#1188] [Maps and Content Showcase]
I've created a simple stats daemon which can get stats from a RE2 server (player count, map name, game mode + modifiers, player names and privileges, etc). It can print JSON to stdout or push the results to an InfluxDB database.

Screenshot of a Grafana instance plotting data from the InfluxDB database (Bandwidth data is from my hypervisor, not this daemon):
![2020-01-04-195357_1440x900_scrot](https://user-images.githubusercontent.com/22487040/71773523-6eab2d80-2f2c-11ea-8724-ac062b0f92e9.png)

If anyone wants to try it out or see the source code, its here: https://github.com/noahb01/re-monitor
I based some of the backend protocol code off of the Redflare source (essentially just ported it to Python3), the Redflare source is here: https://github.com/stainsby/redflare

You'll need Python3.7, python-bitstrings, and python-influxdb if you want to run this yourself. This project is nowhere near complete and I'll be adding features rather frequently.

**Comment by qreeves** [comment]
Please be mindful that collecting any personal information (such as usernames) and storing them exposes you to liability under the GDPR. This is why we were forced to discontinue our own statistics database project.

**Comment by enp2s0** [comment]
As of now individual player names and permissions are dropped before they're put into a database. The closest thing to "user data" that I store is the number of online players, which I think is fine.
Redflare displays current players in it's server list, how do they handle GDPR?

**Comment by enp2s0** [comment]
I'm mostly interested in seeing what maps/gamemodes are most popular, and also what impact a given number of players will have on my server. The daemon pulls player names simply because they're in the datastream that the RE2 server sends; I'll make sure I drop any personal info before it makes it to the database backends.

**Comment by qreeves** [comment]
Redflare doesn't permanently store any data.

**Comment by enp2s0** [comment]
Ah.

I suppose the easiest thing to do then is to just never log any personal info, and limit the database to server stats and not player stats.

**Comment by IceflowRE** [comment]
If you need any info about GDPR, you can ask me, i read the most texts myself and it's pretty "easy".
